src/components/Search/SearchAutocomplete.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import { Autocomplete } from '@material-ui/lab';
+import { createFilterOptions } from '@material-ui/lab/useAutocomplete';
+import TextField from '@material-ui/core/TextField';
+import { Region, State, County } from 'common/regions';
+import MenuItem from './MenuItem';
+
+function getOptionSelected(option: Region, selectedOption: Region) {
+  return option.fipsCode === selectedOption.fipsCode;
+}
+
+const SearchAutocomplete: React.FC<{
+  locations: Region[];
+  filterLimit: number;
+}> = ({ locations, filterLimit }) => {
+  const [input, setInput] = useState('');
+  const [isZip, setIsZip] = useState(false);
+
+  const onInputChange = (e: any, value: string) => {
+    setInput(value);
+    const isStringOfDigits = /^\d+$/.test(value);
+    const isLengthForZipCheck = value.length === 5;
+    if (isStringOfDigits && isLengthForZipCheck) setIsZip(true);
+    else setIsZip(false);
+  };
+
+  const stringifyOption = (option: Region) => {
+    if (isZip) {
+      if ((option as County).zipCodes) {
+        return `${(option as County).zipCodes.join(' ')}`;
+      }
+      return option.name;
+    }
+    return option.name;
+  };
+
+  // Todo (chelsi) - fix this
+  const onSelect = (e: any, value: Region) => {
+    if (value instanceof State) {
+      const stateUrl = (value as State).canonicalUrl;
+      window.location.href = stateUrl;
+    } else if (value instanceof County) {
+      const countyUrl = `${(value as County).state.canonicalUrl}/county/${
+        value.urlSegment
+      }`;
+      window.location.href = countyUrl;
+    }
+    //   else if (value instanceof MetroArea) {
+    //     const metroUrl = `/us/metro/${
+    //       (value as MetroArea).urlSegment
+    //     }`
+    //     window.location.href = metroUrl
+    // }
+    else {
+      window.location.href = '/';
+    }
+  };
+
+  const zipCodeInput = isZip ? input : '';
+
+  return (
+    <Autocomplete
```

Comment:
Should we use the [freeSolo](https://material-ui.com/components/autocomplete/#free-solo) property to remove the triangle? It seems like that's built for our use case: